File: gui/src/editions/commercial.js
// Commercial distribution metadata. The future soulacy-commercial repository
// owns this module and the Teams/Scale implementations it describes.
import { editionCapabilities } from '../lib/edition.js'

export const commercialEditions = [
  {
    id: 'team', displayName: 'Teams', commercial: true, extends: 'personal',
    capabilities: [
      editionCapabilities.multiUser,
      editionCapabilities.workspaceAdmin,
      editionCapabilities.centralizedProviders,
    ],
  },
  {
    id: 'scale', displayName: 'Scale', commercial: true, extends: 'team',
    capabilities: [
      editionCapabilities.externalIdentity,
      editionCapabilities.auditReporting,
      editionCapabilities.distributedRuntime,
      editionCapabilities.highAvailability,
    ],
  },
]

File: internal/distribution/commercial.go
package distribution

import "github.com/soulacy/soulacy/pkg/edition"

// This file is part of the combined repository only. During extraction it
// moves to soulacy-commercial; deleting it leaves a valid Personal registry.
func init() {
	for _, descriptor := range []edition.Descriptor{teamDescriptor(), scaleDescriptor()} {
		if err := Register(descriptor); err != nil {
			panic(err)
		}
	}
}

func teamDescriptor() edition.Descriptor {
	return edition.Descriptor{
		ID: edition.Team, DisplayName: "Teams", Commercial: true, Extends: edition.Personal,
		Capabilities: []edition.Capability{
			edition.MultiUser, edition.WorkspaceAdmin, edition.CentralizedProviders,
		},
	}
}

func scaleDescriptor() edition.Descriptor {
	return edition.Descriptor{
		ID: edition.Scale, DisplayName: "Scale", Commercial: true, Extends: edition.Team,
		Capabilities: []edition.Capability{
			edition.ExternalIdentity, edition.AuditReporting,
			edition.DistributedRuntime, edition.HighAvailability,
		},
	}
}
